### DISCUSSION TITLE
/posts/7-amazing-cli-tools

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#11] [General]
# /posts/7-amazing-cli-tools

These are 7 amazing cli tools that are incredibly helpful when working on the terminal. There's no going back once you try them!

https://www.josean.com/posts/7-amazing-cli-tools

**Comment by gauffer** [comment]
love it! thank you for your hard work!

**Comment by gh0stzk** [comment]
Nice, will add this to my repo. Thank you.

**Comment by asgreen21** [comment]
Love it!

**Comment by vigasin** [comment]
So cool! Thank you!

**Comment by V0R73CKZ** [comment]
Amazing! I have one question though, is there a way to have the lines wrap in the bat preview when Ctrl + T is pressed?

**Comment by markjaquith** [comment]
I already had a bunch of these tools, but I wasn't using them to their full potential. Thanks so much for the level-up! Great video, great post.

**Comment by MemeFormer** [comment]
I feel a bit dumbfounded. I knew almost half of the tools – TLDR and TheFuck – which I've been using for quite some time, but how in the world have I never heard that I can use VSCode to open files? lol. And I was so happy to have found Micro as a replacement for Nano, but I could have had it so much easier – that's crazy.

I think one worthy addition to that already very good list would be [Oh My Bash](https://github.com/ohmybash/oh-my-bash), which is also one you can't go back from once you've tried it.

Having something like [tree](https://osxdaily.com/2016/09/09/view-folder-tree-terminal-mac-os-tree-equivalent/) in Linux is also something I really like.

**Comment by coles980** [comment]
excelentes tools :)

**Comment by Nazaroni** [comment]
Amazing video and blog post! Thank you for all of us for helping us to be more productive and efficient! Respect!

**Comment by ph-cardoso** [comment]
Amazing work!

**Comment by m1st0** [comment]
Amazed at the well done direction and creativity.  I needed to try some of these things on Ubuntu and Bash, so here is my script gist [fzf.bash](https://gist.github.com/m1st0/70e25d3f265e9fcec4c6c36a4e650fdb) .

**Reply by m1st0** [reply]
. . . with instructions.

**Comment by gh0stzk** [comment]
What if i dont wanna use the ** to trigger fzf?? what if i just wanna press tab?

**Comment by praveensinghaws** [comment]
Lovely

**Comment by hdsenevi** [comment]
Mate, this is super helpful. Amazing work. Keep it up. 👏

**Comment by cookiebinary1** [comment]
Great!!!
My favourite other tool is ZSH autosuggestion:
```bash
brew install zsh-autosuggestions
```
Src: https://github.com/zsh-users/zsh-autosuggestions/

**Comment by shahjalal-labs** [comment]
hey joss,, tnx brother

**Comment by mtennes** [comment]
I have a quick question:

When I run 
```
source ~/.zshrc
```
I get many of the following errors
```
compinit:527: no such file or directory: /usr/local/homebrew/share/zsh/site-functions/_ykman
compinit:shift:529: shift count must be <= $#
```
Seems like the following are symlinks and maybe that's breaking things.
```
❯ ~ ls -la /usr/local/Homebrew/share/zsh/site-functions
--  _brew -> ../../../completions/zsh/_brew
-I  _brew_services -> ../../../Library/Taps/homebrew/homebrew-services/completions/zsh/_brew_services
-I  _keyring -> ../../../Cellar/keyring/24.3.0/share/zsh/site-functions/_keyring
-I  _lf -> ../../../Cellar/lf/31/share/zsh/site-functions/_lf
-I  _meson -> ../../../Cellar/meson/1.2.3/share/zsh/site-functions/_meson
-I  _ninja -> ../../../Cellar/ninja/1.11.1/share/zsh/site-functions/_ninja
-I  _rg -> ../../../Cellar/ripgrep/13.0.0/share/zsh/site-functions/_rg
-I  _tldr -> ../../../Cellar/tldr/1.6.0/share/zsh/site-functions/_tldr
```
**Any help would be greatly appreciated**

**Reply by mtennes** [reply]
and
1```
-I  _ykman -> ../../../Cellar/ykman/5.2.1_1/share/zsh/site-functions/_ykman
```

**Reply by m1st0** [reply]
In case you need this still for YubiKey, which seems to be missing:
`brew reinstall yk-manager`

1. Delete the stale cache file(s)
`rm ~/.zcompdump*`

2. Force Zsh to rebuild the cache
`autoload -Uz compinit && compinit`

3. `source ~/.zshrc` or open a new terminal. Hopefully your ~/.zshrc file contains a line that automatically calls `compinit` when a new terminal opens.

**Comment by daylight-roberty** [comment]
You have no idea how much this improved my life over the last 6 months

**Comment by roshanmelanta** [comment]
Loved it. Thank you so much!!
